src/gui/processing/qgsprocessingmodelerwidget.cpp

```diff
@@ -0,0 +1,290 @@
+  connect( mSourceMenu, &QMenu::triggered, this, &QgsProcessingModelerParameterWidget::sourceMenuActionTriggered );
+  mSourceButton->setMenu( mSourceMenu );
+
+  hLayout->addWidget( mSourceButton );
+
+  mStackedWidget = new QStackedWidget();
+
+  mStaticWidgetWrapper.reset( QgsGui::processingGuiRegistry()->createParameterWidgetWrapper( mParameterDefinition, QgsProcessingGui::Modeler ) );
+  if ( mStaticWidgetWrapper )
+  {
+    QWidget *widget = mStaticWidgetWrapper->createWrappedWidget( context );
+    if ( widget )
+    {
+      mHasStaticWrapper = true;
+      mStackedWidget->addWidget( widget );
+    }
+    else
+      mStackedWidget->addWidget( new QWidget() );
+  }
+  else
+  {
+    mStackedWidget->addWidget( new QWidget() );
+  }
+
+  mExpressionWidget = new QgsExpressionLineEdit();
+  mStackedWidget->addWidget( mExpressionWidget );
+
+  mModelInputCombo = new QComboBox();
+  QHBoxLayout *hLayout2 = new QHBoxLayout();
+  hLayout2->setMargin( 0 );
+  hLayout2->setContentsMargins( 0, 0, 0, 0 );
+  hLayout2->addWidget( new QLabel( tr( "Using model input" ) ) );
+  hLayout2->addWidget( mModelInputCombo, 1 );
+  QWidget *hWidget2 = new QWidget();
+  hWidget2->setLayout( hLayout2 );
+  mStackedWidget->addWidget( hWidget2 );
+
+  mChildOutputCombo = new QComboBox();
+  QHBoxLayout *hLayout3 = new QHBoxLayout();
+  hLayout3->setMargin( 0 );
+  hLayout3->setContentsMargins( 0, 0, 0, 0 );
+  hLayout3->addWidget( new QLabel( tr( "Using algorithm output" ) ) );
+  hLayout3->addWidget( mChildOutputCombo, 1 );
+  QWidget *hWidget3 = new QWidget();
+  hWidget3->setLayout( hLayout3 );
+  mStackedWidget->addWidget( hWidget3 );
+
+  hLayout->setMargin( 0 );
+  hLayout->setContentsMargins( 0, 0, 0, 0 );
+  hLayout->addWidget( mStackedWidget, 1 );
+
+  setLayout( hLayout );
+  setSourceType( QgsProcessingModelChildParameterSource::StaticValue );
+}
+
+QgsProcessingModelerParameterWidget::~QgsProcessingModelerParameterWidget() = default;
+
+const QgsProcessingParameterDefinition *QgsProcessingModelerParameterWidget::parameterDefinition() const
+{
+  return mParameterDefinition;
+}
+
+QLabel *QgsProcessingModelerParameterWidget::createLabel()
+{
+  if ( mStaticWidgetWrapper )
+    return mStaticWidgetWrapper->createWrappedLabel();
+  else
+    return nullptr;
+}
+
+void QgsProcessingModelerParameterWidget::setWidgetValue( const QgsProcessingModelChildParameterSource &value )
+{
+  // we make a copy of all attributes and store locally, so that users can flick between
+  // sources without losing their current value
+  mStaticValue = value.staticValue();
+  mModelInputParameterName = value.parameterName();
+  mOutputChildId = value.outputChildId();
+  mOutputName = value.outputName();
+  mExpression = value.expression();
+
+  updateUi();
+  setSourceType( value.source() );
+}
+
+QgsProcessingModelChildParameterSource QgsProcessingModelerParameterWidget::value() const
+{
+  switch ( mStackedWidget->currentIndex() )
+  {
+    case 0:
```

Comment:
The current index is frequently used in this class. Do you think that an enum could be defined instead?

Reply:
Did you forget this comment due to the file renaming, or is it just irrelevant in this case?

Reply:
Forgot sorry! Will address

Reply:
Nice, thanks :)